File: backend/models/credential.go
package models

import (
	"time"
)

type Credential struct {
	ID          uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Name        string    `gorm:"type:varchar(128);not null" json:"name"`
	Username    string    `gorm:"type:varchar(128);not null" json:"username"`
	Password    string    `gorm:"type:varchar(512);not null" json:"-"`
	EnablePW    string    `gorm:"type:varchar(512)" json:"-"`
	SSHKey      string    `gorm:"type:text" json:"-"`
	AuthMethod  string    `gorm:"type:varchar(16);default:password" json:"auth_method"`
	Description string    `gorm:"type:varchar(256)" json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

func (Credential) TableName() string {
	return "credentials"
}

File: frontend/src/api/credential.ts
import request from './request'

export interface Credential {
  id: number
  name: string
  username: string
  auth_method: string
  description: string
  created_at: string
  updated_at: string
}

export function getCredentials() {
  return request.get<Credential[]>('/credentials')
}

export function getCredential(id: number) {
  return request.get<Credential>(`/credentials/${id}`)
}

export function createCredential(data: {
  name: string
  username: string
  password: string
  enable_pw?: string
  auth_method?: string
  description?: string
}) {
  return request.post<Credential>('/credentials', data)
}

export function updateCredential(id: number, data: Partial<Credential>) {
  return request.put(`/credentials/${id}`, data)
}

export function deleteCredential(id: number) {
  return request.delete(`/credentials/${id}`)
}
